### PR TITLE
feat: Rename environment variables for the sake of consistency

### LINKED ISSUE #399: Environment Variable Naming Consistency
# 🚀 Tech Debt Request

### Relevant Package [**REQUIRED**]

 This feature request is for environment variables

### Description [**REQUIRED**]
 The names use `CONFIG`, `CONF`, and `CONFIGURATION` inconsistently.
```
	envKeyConfigUrl       = "EDGEX_CONFIGURATION_PROVIDER"
	envConfDir            = "EDGEX_CONF_DIR"
	envFile               = "EDGEX_CONFIG_FILE"
```

### Describe the solution you'd like
 If you have a solution in mind, please describe it.

```
	envConfigProviderUrl       = "EDGEX_CONFIG_PROVIDER"
	envConfigDir            = "EDGEX_CONFIG_DIR"
	envConfigFile               = "EDGEX_CONFIG_FILE"
```
Note that the command line options for these will need to be adjusted accordingly.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by FelixTing**
resolve #399 

BREAKING CHANGE:
- `EDGEX_CONFIGURTION_PROVIDER` is replaced by `EDGEX_CONFIG_PROVIDER`
- `EDGEX_CONF_DIR` is replaced by `EDGEX_CONFIG_DIR`

Signed-off-by: Felix Ting <felix@iotechsys.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?) https://github.com/edgexfoundry/edgex-docs/pull/914

## Testing Instructions

1. Download this docker-compose file(https://github.com/edgexfoundry/edgex-compose/blob/main/docker-compose-no-secty.yml) and edit it to remove device-virtual
2. Run EdgeX Foundry with the docker-compose file
3. Download https://github.com/FelixTing/go-mod-bootstrap/tree/issue-399
4. Download device-virtual (https://github.com/edgexfoundry/device-virtual-go), edit go.mod to replace github.com/edgexfoundry/go-mod-bootstrap/v2 with the one downloaded in step 3. For example
```
replace github.com/edgexfoundry/go-mod-bootstrap/v2 => ../go-mod-bootstrap
```
5. Build a binary executable for device-virtual
```
make build
```
6. Set environment variables, for example:
```
export EDGEX_CONFIG_PROVIDER=consul.http://localhost:8500
export EDGEX_CONFIG_FILE=my-config.toml
```
7. Run device-virtual
8. Logs similar to the following are expected to appear:
```
level=INFO ts=2022-12-07T04:43:41.025805Z app=device-virtual source=variables.go:377 msg="Variables override of '-f/-file' by environment variable: EDGEX_CONFIG_FILE=my-config.toml"
level=INFO ts=2022-12-07T04:43:41.029159Z app=device-virtual source=variables.go:377 msg="Variables override of 'Configuration Provider Information' by environment variable: EDGEX_CONFIG_PROVIDER=consul.http://localhost:8500"

```

### CODE REVIEW

File: bootstrap/environment/variables.go [hunk 1]
```diff
@@ -36,11 +36,11 @@ const (
 	bootRetrySecondsDefault   = 1
 	defaultConfDirValue       = "./res"
 
-	envKeyConfigUrl       = "EDGEX_CONFIGURATION_PROVIDER"
+	envKeyConfigUrl       = "EDGEX_CONFIG_PROVIDER"
 	envKeyUseRegistry     = "EDGEX_USE_REGISTRY"
 	envKeyStartupDuration = "EDGEX_STARTUP_DURATION"
 	envKeyStartupInterval = "EDGEX_STARTUP_INTERVAL"
-	envConfDir            = "EDGEX_CONF_DIR"
+	envConfDir            = "EDGEX_CONFIG_DIR"
```
Comment (ejlee3) [inline bootstrap/environment/variables.go:43]:
```suggestion
	envConfigDir            = "EDGEX_CONFIG_DIR"
```

File: bootstrap/environment/variables.go [hunk 2]
```diff
@@ -36,11 +36,11 @@ const (
...
 	envProfile            = "EDGEX_PROFILE"
 	envFile               = "EDGEX_CONFIG_FILE"
```
Comment (ejlee3) [inline bootstrap/environment/variables.go:45]:
```suggestion
	envConfigFile               = "EDGEX_CONFIG_FILE"
```

**Comment by ejlee3** [comment]
Commit message should be `refactor!:` instead of `feat:`. This really is a refactor of the code and the `!` helps denote the breaking change.

**Comment by cloudxxx8** [comment]
> I think we should also rename the corresponding command line flags
> 
> * `-c/--confdir` to `-cd/--configDir`
> * `-f/--file` to `-cf/--configFile`

the issue https://github.com/edgexfoundry/edgex-go/issues/4248 is opened for this, so we won't make the related change in this PR

**Review by ejlee3 — APPROVED** [review]
LGTM - I was able to see the changes with the testing instructions.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**